### PR TITLE
[tree widget] Changes for blocklisting

### PR DESCRIPTION
**Pull request by chejennifer**
- update cache to have a search index that doesn't include blocklisted values
- update SearchStatVarRequest to take in an additional enable_blocklist boolean. If this boolean is true, use search index that doesn't include blocklisted values
- revert change that set the numDescendents for the SVG Employment

### CODE REVIEW

File: internal/server/statvar_hierarchy_util.go [hunk 1]
```diff
@@ -48,6 +48,8 @@ type SearchIndex struct {
 // PVs to a number greater than max number of PVs for a human curated stat var.
 const nonHumanCuratedNumPv = 30
 
+var blocklistedSvgIds = []string{"dc/g/Person_EmploymentStatus", "dc/g/Establishment", "dc/g/Person_Industry"}
```
Comment (shifucun) [inline internal/server/statvar_hierarchy_util.go:51]: please add a comment that this should be synced with website repo

File: internal/server/statvar_hierarchy_util.go [hunk 2]
```diff
@@ -123,16 +125,40 @@ func (index *SearchIndex) update(
...
 }
 
+// Helper to build a set of ignored SVGs.
+func getIgnoredSVGHelper(
+	ignoredSvg map[string]string,
+	rawSvg map[string]*pb.StatVarGroupNode,
+	svgID string) {
+	ignoredSvg[svgID] = ""
+	svgData, ok := rawSvg[svgID]
+	if ok {
```
Comment (shifucun) [inline internal/server/statvar_hierarchy_util.go:135]: nit: can write in one line as:

if svgData, ok := rawSvg[svgID]; ok {
...
}

File: internal/server/statvar_search.go [hunk 3]
```diff
@@ -34,6 +34,7 @@ func (s *Server) SearchStatVar(
 ) {
 	query := in.GetQuery()
 	places := in.GetPlaces()
+	blocklist := in.GetEnableBlocklist()
```
Comment (shifucun) [inline internal/server/statvar_search.go:37]: nit: naming "blocklist" -> "enableBlocklist"

**Review by shifucun — APPROVED** [review]
Thanks for the quick change!

File: internal/server/statvar_hierarchy_util.go (hunk 1 above)
Comment (chejennifer) [reply internal/server/statvar_hierarchy_util.go:52]: done

File: internal/server/statvar_hierarchy_util.go (hunk 2 above)
Comment (chejennifer) [reply internal/server/statvar_hierarchy_util.go]: done

File: internal/server/statvar_search.go (hunk 3 above)
Comment (chejennifer) [reply internal/server/statvar_search.go]: done

**Pull request merged**